src/lib.rs

```diff
@@ -498,14 +516,26 @@ impl Cmd {
     fn output_impl(&self, read_stdout: bool, read_stderr: bool) -> io::Result<Output> {
         let mut child = {
             let _guard = gsl::read();
-            self.command()
-                .stdin(match &self.stdin_contents {
-                    Some(_) => Stdio::piped(),
-                    None => Stdio::null(),
-                })
-                .stdout(if read_stdout { Stdio::piped() } else { Stdio::inherit() })
-                .stderr(if read_stderr { Stdio::piped() } else { Stdio::inherit() })
-                .spawn()?
+            let mut command = self.command();
+
+            command.stdin(match &self.stdin_contents {
+                Some(_) => Stdio::piped(),
+                None => Stdio::null(),
+            });
+
+            if read_stdout && !self.ignore_stdout {
+                command.stdout(Stdio::piped());
+            } else if !read_stdout && !self.ignore_stdout {
+                command.stdout(Stdio::inherit());
+            }
```

Comment:
```suggestion
            if !self.ignore_stdout {
                command.stdout(if read_stdout  { Stdio::piped() } else { Stdio::inherit() });
            }
```

I think this might be a bit more readable, as it factors out the common `!self.ignore_stdout` condition.